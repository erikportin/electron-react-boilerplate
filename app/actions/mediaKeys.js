// @flow
export const PLAY_PAUSE_ACTIVATED = 'PLAY_PAUSE_ACTIVATED';
export const PLAY_PAUSE_DEACTIVATED = 'PLAY_PAUSE_DEACTIVATED';

type actionType = {
  +type: string
};

export function activate() {
  return {
    type: PLAY_PAUSE_ACTIVATED
  };
}

export function deactivate() {
  return {
    type: PLAY_PAUSE_DEACTIVATED
  };
}
