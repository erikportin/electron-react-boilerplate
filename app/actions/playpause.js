// @flow
import type { counterStateType } from '../reducers/counter';

type actionType = {
  +type: string
};

export const PLAY_PAUSE = 'PLAY_PAUSE';

export function playPause() {
  return {
    type: PLAY_PAUSE
  };
}