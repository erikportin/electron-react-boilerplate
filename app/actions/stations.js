// @flow
export const ADD_STATION = 'ADD_STATION';
export const REMOVE_STATION = 'REMOVE_STATION';

type actionType = {
  +type: string
};

export function add() {
  return {
    type: ADD_STATION
  };
}

export function remove() {
  return {
    type: REMOVE_STATION
  };
}
