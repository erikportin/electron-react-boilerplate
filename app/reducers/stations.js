import { persistentReducer } from 'redux-pouchdb';
import { REMOVE_STATION, ADD_STATION } from '../../app/actions/stations';

export type stationStateType = {
  name: url,
  url: string,
  script: string
};

export type stationsStateType = {
  +stations: Array<stationStateType>
};

type actionType = {
  +type: string
};

export const initialState = [
  {
    url: 'http://www.bbc.co.uk/programmes/b09s37vt',
    name: 'BBC 6',
    script: `var iFrame = document.querySelectorAll('.episode-playout iframe')[0].contentDocument;
    var pauseBtn = iFrame.querySelectorAll('.p_pauseButton')[0];
    var playBtn = iFrame.querySelectorAll('.p_playButton')[0];
    var startBtn = iFrame.querySelectorAll('.p_button')[0];
    
    
    if(pauseBtn){
        pauseBtn.click()
    }
    else if(playBtn){
        playBtn.click()
    }
    else if(startBtn){
        startBtn.click()
    }`
  }
];

const stations = (state = initialState, action) => {
  switch (action.type) {
    case ADD_STATION:
      return [
        ...state,
        action.station
      ];
    case REMOVE_STATION:
      return [
        ...state.slice(0, action.id),
        ...state.slice(action.id + 1)
      ];
    default:
      return state;
  }
};

export default persistentReducer(stations);
