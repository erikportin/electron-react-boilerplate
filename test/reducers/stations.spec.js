import stations from '../../app/reducers/stations';
import { REMOVE_STATION, ADD_STATION } from '../../app/actions/stations';

describe('stations', () => {
  it('should handle initial state', () => {
    expect(stations(undefined, {})).toMatchSnapshot();
  });

  it('should add stations', () => {
    let stationsArr = stations([], {
      type: ADD_STATION,
      station: {
        url: 'url.to.station',
        name: 'Name of station',
        script: 'SCRIPT'
      }
    });
    expect(stationsArr.length).toEqual(1);
    stationsArr = stations(stationsArr, {
      type: ADD_STATION,
      station: {
        url: 'url.to.another.station',
        name: 'Name of another station',
        script: 'ANOTHER SCRIPT'
      }
    });
    expect(stationsArr.length).toEqual(2);
    expect(stationsArr[0].name).toEqual('Name of station');
    expect(stationsArr[1].name).toEqual('Name of another station');
  });

  it('should remove stations', () => {
    let stationsArr = stations([
      {
        url: 'url.to.another.station',
        name: 'Name of another station',
        script: 'ANOTHER SCRIPT'
      },
      {
        url: 'url.to.another.station',
        name: 'Name of another station',
        script: 'ANOTHER SCRIPT'
      }
    ], {
      type: REMOVE_STATION,
      id: 1
    });
    expect(stationsArr.length).toEqual(1);

    stationsArr = stations(stationsArr, {
      type: REMOVE_STATION,
      id: 0
    });
    expect(stationsArr.length).toEqual(0);
  });
});
