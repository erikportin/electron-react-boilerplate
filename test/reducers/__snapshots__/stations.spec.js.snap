// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stations should handle initial state 1`] = `
Array [
  Object {
    "name": "BBC 6",
    "script": "var iFrame = document.querySelectorAll('.episode-playout iframe')[0].contentDocument;
    var pauseBtn = iFrame.querySelectorAll('.p_pauseButton')[0];
    var playBtn = iFrame.querySelectorAll('.p_playButton')[0];
    var startBtn = iFrame.querySelectorAll('.p_button')[0];
    
    
    if(pauseBtn){
        pauseBtn.click()
    }
    else if(playBtn){
        playBtn.click()
    }
    else if(startBtn){
        startBtn.click()
    }",
    "url": "http://www.bbc.co.uk/programmes/b09s37vt",
  },
  Object {
    "name": "SR.se",
    "script": "document.getElementsByClassName(\\"player-play\\")[0].click()",
    "url": "http://sr.se",
  },
]
`;
